### DISCUSSION TITLE
d64 support

### DISCUSSION THREAD

**Discussion by Marc91360** [frntc/RAD#57] [General]
Hello,

Is there a chance that one day the RAD may support d64 image ?
If I'm not wrong, only PRG are supporter right now.
If I ask the question, it's because I have an idea in mind :)
Do you know the excellent "VISION Basic" project? A new, powerful Basic that requires memory expansion ...
The RAD will work perfecly with it :)

Best regards from France,

**Comment by frntc** [comment]
Hi,

what kind of support do you have in mind? Replacing kernal vectors to support LOAD, SAVE etc. is the easiest way to provide some support, but with little compatibility compared to IEC-based devices... is that enough?

**Comment by Marc91360** [comment]
Hello,I was thinking about something that can allow to « mount » a d64 image - and then load files from it.Right now we can only launch PRG files, REU, if I’m not wrong, using the browser menu, but d64 image are not supported ?Best regards,MarcEnvoyé de mon iPhoneLe 25 sept. 2024 à 21:55, frntc ***@***.***> a écrit :﻿
Hi,
what kind of support do you have in mind? Replacing kernal vectors to support LOAD, SAVE etc. is the easiest way to provide some support, but with little compatibility compared to IEC-based devices... is that enough?

**Comment by tinspin** [comment]
What about .crt, is that hard to support on RAD?

**Reply by frntc** [reply]
That's possible via emulating the 6510 and the PLA. I have this running, but not yet in release condition...

**Comment by tomnic79** [comment]
I'd like to use only RAD to run Sonic the Hedgehog game, so *.d64 or much better *.d81 images should be loaded. Do you think it is possible to convert *.d81 to *.crt and then *.crt to *.prg which works right now with RAD?

**Comment by futureretrofusion** [comment]
Most of the loading in Sonic c64 is into the REU before the game starts.

It could be done I guess with a .prg if you had the game data already in a
16mb  .reu bank...

**Comment by frntc** [comment]
I'm sure it would be possible to create a PRG+REU-image version (but I think this was not planned for Sonic). Converting D81->CRT->PRG is something that will not work (the D81 holds much more data than a PRG)

**Comment by frntc** [comment]
The VSF-loading of the new firmware can load a Vice snapshot created when Sonic reached its main menu. Might be a solution to your question ;-)

**Comment by Marc91360** [comment]
Hello all !
I'm very happy that the new firmware allow to load Vice snapshot :) 
But my original question was about a ".d64 support", not about a turn around solution for the (great) game Sonic.

Best regards;

Myself

**Reply by frntc** [reply]
Yes, your questions was hijacked in between ;-) While possible (rather at the level of kernal vectors) it's something that would cost quite some time to implement, and free time is a scarce resource atm :-/

**Comment by futureretrofusion** [comment]
This is a genius and simple update.

So I can load custom kernals..(by making Snapshots of them) they should
work fine as they are injected in ram on boot normally?

Is it possible to modify the reset sequence so that the kernel keeps
getting loaded on reset... The answer is probably in your docs 😎

But amazing,  I have globe demo running instantly. A snapshot of Super
Copyright Bros works great as well.

Is there any code or setting  that could allow the management of the multi
load carts in memory? Ie easy flash Snapshots fail at the point they try to
access more data...

I'll eventually tinker myself.

Thank you so much for everything you've done.

**Reply by frntc** [reply]
You're asking about something that I wanted to add to the description: snapshots with different kernals than in the real machine will typically crash, because some vectors in RAM might assume to find something in the ROM (which is different).

Right, once an EF- or MD-CRT will try to access some more data it'll crash because that hardware is not emulated. It's really just dumping the snapshot to RAM once and roughly setting the timers and VIC-sync before starting. I assume most demos/intros will probably not work via VSFs because the timings are not exact. Also the disk status is not restored (simply because I have never even thought how this would be done).

**Comment by futureretrofusion** [comment]
I think that regardless of minor limitations, so much more can now be
loaded and works.

It's like a flash cart now with REU.

This hugely expands the usability and functionality of the RAD.
